### DISCUSSION TITLE
Allow user to change Email issue

### DISCUSSION THREAD

**Discussion by dev13-suthar** [lukevella/rallly#1399] [Ideas]
These is my approach for Solving issue #626 .
![Screenshot 2024-10-17 121538](https://github.com/user-attachments/assets/df9b9356-5be9-4f17-bd06-92141dacd879)
open for other improvments.
should i go with these @lukevella

**Comment by lukevella** [comment]
Hey @dev13-suthar, I don't recommend picking this up. It's too large of a task and I don't have capacity to oversee and review this work right now.

**Reply by dev13-suthar** [reply]
Sure, We can take these up whenever u free, just ping me, I Have better approach then above and i have implemented it in my project also.

**Reply by lukevella** [reply]
Thanks @dev13-suthar. I've already put some thought into this and  it appears to be a pretty big piece of work since it involves updating the UI, auth, database schema as well as running a database migration and handling a number of edge cases. It would require a lot of attention. Feel free to share examples of how you've implemented this.

**Comment by dev13-suthar** [comment]
got little busy
So these Are Genral Steps:
->Take input of new Email From User and user will click save,
->on Server we will generate unique/encypted token with secret_key and user email and save that token in userDb, Send mail to users orignal mail with url /changeEmail?token={encyptedToken_we_saved_in_User_Db}.
->User will click Change/Verify Email, on These click we will perform Verification with the Params token and Users DB token with expiryTime  and we will compare both if they are correct we will decrypt token to user Email and save that Email in Db.
